aws/data_source_aws_lex_intent.go

```diff
@@ -0,0 +1,83 @@
+package aws
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/lexmodelbuildingservice"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func dataSourceAwsLexIntent() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsLexIntentRead,
+
+		Schema: map[string]*schema.Schema{
```

Comment:
Is there a reason why you chose to not export all of the properties associated with an intent?

Reply:
the intent `arn` would be good to build `aws_lambda_permission` policies to give the intent permission to call a lambda, here's an example of the resulting policy (the `*` is the `version`)

```
  {
      "Sid": "lex-us-east-1-AgentSupport",
      "Effect": "Allow",
      "Principal": {
        "Service": "lex.amazonaws.com"
      },
      "Action": "lambda:invokeFunction",
      "Resource": "arn:aws:lambda:us-east-1:449704592230:function:EmployeeBooking_LexBookHotel",
      "Condition": {
        "ArnLike": {
          "AWS:SourceArn": "arn:aws:lex:us-east-1:449704592230:intent:AgentSupport:*"
        }
      }
    }
```

Reply:
Composing the ARN (as suggested by gdavison) and adding it to the resource and data source now.